internal/uvm/create.go

```diff
@@ -66,6 +76,53 @@ type Options struct {
 	ExternalGuestConnection bool
 }
 
+// Verifies that the final UVM options are correct and supported.
+func verifyOptions(ctx context.Context, options interface{}) error {
```

Comment:
Was originally going to do this in https://github.com/microsoft/hcsshim/pull/809 and call this at the end of SpecToUVMCreateOpts. Anyone have a preference? I feel the errs should be constants and we can add more if the amount of unsupported/wrong configs grows.